### PR TITLE
Testing workflow setup

### PR DESCRIPTION
**Pull request by piyusharmap**
### 📃 Description

Added workflow for testing and remove take issue worflow.

**Related Tickets & Documents**

Fixes # (issue)

**What type of PR is this? (check all applicable)**

- [x] 💡 Feature
- [ ] 🐛 Bug Fix
- [ ] 📃 Documentation Update
- [ ] 🎨 Style Changes
- [ ] 💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test

### 🔰 Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### ❓ How Has This Been Tested?

NIL

### 🔰 Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  run-tests:
```
Comment (TanishqSingla) [inline .github/workflows/test.yml:11]: Make this workflow depend on lint workflow

**Review by TanishqSingla — CHANGES_REQUESTED** [review]

**Comment by TanishqSingla** [comment]
Looks good to me, can you please evaluate test runners like jest or vitest and include the appropriate one in the PR. Also include a sample test case that way we can confirm the pipeline is running

**Comment by piyusharmap** [comment]
Got it, will do

File: src/__tests__/purifyHTML.test.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { purifyHTMLString } from '@/utils/purifyHTML';
+import { describe, expect, it } from 'vitest';
+
+describe('purifyHTMLString', () => {
+  it('returns the same string if it is clean', () => {
+    const cleanString = 'Hello, world!';
+    expect(purifyHTMLString(cleanString)).toBe(cleanString);
```
Comment (TanishqSingla) [inline src/__tests__/purifyHTML.test.ts:7]: Follow AAA (Arrange Act Assert) convention

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      - if: ${{ steps.cache-npm.outputs.cache-hit != 'true' }}
+        name: List the state of node modules
+        continue-on-error: true
+        run: npm list
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Run test
+        run: npm run test || true
```
Comment (TanishqSingla) [inline .github/workflows/test.yml:46]: Fail the workflow if test fails

File: src/__tests__/purifyHTML.test.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    expect(purifyHTMLString(dirtyString)).toBe('');
+  });
+
+  it('handles an empty string gracefully', () => {
+    expect(purifyHTMLString('')).toBe('');
+  });
+
+  it('throws an error for non-string inputs', () => {
+    // @ts-expect-error Testing non-string input
+    expect(() => purifyHTMLString(null)).toThrow();
```
Comment (TanishqSingla) [inline src/__tests__/purifyHTML.test.ts:21]: Follow AAA convention for this

**Review by TanishqSingla — COMMENTED** [review]
LGTM, only got convention issues for now. Let's go with this for now and will update this as per requirements

**Comment by TanishqSingla** [comment]
Also add test coverage metrics, we need them going forward

File: src/__tests__/purifyHTML.test.ts (hunk 2 above)
Comment (piyusharmap) [reply src/__tests__/purifyHTML.test.ts:7]: Not very familiar with the AAA convention, but I will study it and implement

File: .github/workflows/test.yml (hunk 3 above)
Comment (piyusharmap) [reply .github/workflows/test.yml]: done

**Comment by piyusharmap** [comment]
Don't have much knowledge about test coverage metrics, but I will discuss further and add them later.

**Review by TanishqSingla — APPROVED** [review]

**Comment by TanishqSingla** [comment]
LGTM for now. Let's go with this

**Pull request merged**